### PR TITLE
Upgrade Dredd Transactions and refactor how annotations are processed

### PR DESCRIPTION
**Pull request by honzajavorek**
#### :rocket: Why this change?

Dredd Transactions contain a lot of new fixes and features (not only) in OAS3 support. There's more and more issues on Dredd which are filed only because Dredd is behind in what has been already developed.

Dredd is behind because new versions of the parsing toolchain caught me off guard during refactoring rather larger portions of the Dredd codebase. This should be the last missing piece, which will address pending issues around annotation line/column numbers and will upgrade Dredd Transactions in the process to address outstanding OAS3 issues. 

#### :memo: Related issues and Pull Requests

- Close https://github.com/apiaryio/dredd/issues/127 - @cw-peter & @tunovic please verify your issue is addressed when this gets merged & released 🙏 
- Close https://github.com/apiaryio/dredd/issues/916
- Close https://github.com/apiaryio/dredd/issues/1191
- Close https://github.com/apiaryio/dredd/issues/1287 - @y-luis please verify your issue is addressed when this gets merged & released 🙏 
- Close https://github.com/apiaryio/dredd/issues/1334 - @yourithielen please verify your issue is addressed when this gets merged & released 🙏 
- Close https://github.com/apiaryio/dredd/issues/1307 - @simondutertre please verify your issue is addressed when this gets merged & released 🙏 

#### :white_check_mark: What didn't I forget?

- [ ] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

**Comment by honzajavorek** [comment]
Getting closer 🛠 I need to call it a day now but only two tests are failing.

Screenshot 2019-05-07 at 13 36 27

File: lib/Dredd.js [hunk 1]
```diff
@@ -165,13 +173,19 @@ class Dredd {
     this.logger.debug('Preparing API description documents');
     this.prepareAPIdescriptions((error, apiDescriptions) => {
       if (error) { callback(error, this.stats); return; }
-      this.configuration.apiDescriptions = apiDescriptions;
 
-      // TODO https://github.com/apiaryio/dredd/issues/1191
-      const annotationsError = handleRuntimeProblems(apiDescriptions, this.logger);
-      if (annotationsError) { callback(annotationsError, this.stats); return; }
+      const loggerInfos = toLoggerInfos(apiDescriptions);
+      // FIXME: Winston 3.x supports calling .log() directly with the loggerInfo
```
Comment (artem-zakharchenko) [inline lib/Dredd.js]: We can reference https://github.com/apiaryio/dredd/issues/1225 here to keep track of the changes.

File: lib/compileTransactionName.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// see https://github.com/apiaryio/dredd-transactions/issues/276
+
+
+module.exports = function compileTransactionName(origin) {
+  const segments = [];
+  if (origin.apiName) { segments.push(origin.apiName); }
+  if (origin.resourceGroupName) { segments.push(origin.resourceGroupName); }
+  if (origin.resourceName) { segments.push(origin.resourceName); }
+  if (origin.actionName) { segments.push(origin.actionName); }
+  if (origin.exampleName) { segments.push(origin.exampleName); }
```
Comment (artem-zakharchenko) [inline lib/compileTransactionName.js]: I understand this is copy-pasted, but you can also do a slightly shorter syntax:

```js
const segments = [
  origin.apiName,
  origin.resourceGroupName,
  origin.resourceName,
  origin.actionName,
  origin.exampleName,
].filter(Boolean).join(' > ')
```

Apart from shorter declaration it also reduces the function's complexity.

File: lib/Dredd.js (hunk 1 above)
Comment (honzajavorek) [reply lib/Dredd.js]: Added

File: lib/compileTransactionName.js (hunk 2 above)
Comment (honzajavorek) [reply lib/compileTransactionName.js]: That's a good suggestion. I'll file it as a DT issue and fix it there. This wile stays copy-pasted and then deleted once DT provides the name directly.

Comment (honzajavorek) [reply lib/compileTransactionName.js]: https://github.com/apiaryio/dredd-transactions/issues/283

**Pull request merged**